datastore/__init__.py

```diff
@@ -661,6 +666,87 @@ async def get_seed_info(cursor):
         seed_dict["seeds"].append({"seed_id": seed_id, "seed_name": seed_name})
     return seed_dict
 
+async def get_picture_sets_info(cursor, user_id: str):
+    """This function retrieves the picture sets names and number of pictures from the database.
+
+    Args:
+        user_id (str): id of the user
+    """
+    # Check if user exists
+    if not user.is_a_user_id(cursor=cursor, user_id=user_id):
+        raise user.UserNotFoundError(
+            f"User not found based on the given id: {user_id}"
+        )
+    
+    result = {}
+    picture_sets = picture.get_user_picture_sets(cursor, user_id)
+    for picture_set in picture_sets:
+        picture_set_id = picture_set[0]
+        picture_set_name = picture_set[1]
+        nb_picture = picture.count_pictures(cursor, picture_set_id)
+        result[picture_set_id] = [picture_set_name, nb_picture]
+    return result
+
+async def delete_picture_set(cursor, user_id, picture_set_id, container_client):
+    """
+    Delete a picture set from the database and the blob storage
+
+    Args:
+        cursor: The cursor object to interact with the database.
+        user_id (str): id of the user
+        picture_set_id (str): id of the picture set to delete
+        container_client: The container client of the user.
+    """
+    try:
+        # Check if user exists
+        if not user.is_a_user_id(cursor=cursor, user_id=user_id):
+            raise user.UserNotFoundError(
+                f"User not found based on the given id: {user_id}"
+            )
+        # Check if picture set exists
+        if not picture.is_a_picture_set_id(cursor, picture_set_id):
+            raise picture.PictureSetNotFoundError(
+                f"Picture set not found based on the given id: {picture_set_id}"
+            )
+        # Check user is owner of the picture set
+        if not picture.get_picture_set_owner_id(cursor, picture_set_id) == user_id:
+            raise picture.PictureSetDeleteError(
+                f"User can't delete this folder, user uuid :{user_id}, folder name : {picture_set_id}"
+            )
+        # Check if the picture set is the default picture set
+        general_folder_id = user.get_default_picture_set(cursor, user_id)
+        if general_folder_id == picture_set_id:
+            raise picture.PictureSetDeleteError(
+                f"User can't delete the default picture set, user uuid :{user_id}"
+            )
+
+        if picture.count_pictures(cursor, picture_set_id) > 0:
+            #move the pictures to the default picture set	
```

Comment:
I dont get why we move the pictures? Also what happens if the user deletes the General folder?

Reply:
User can't delete it because of the verification at line 718, and we move the pictures to save the inferences but maybe we find another way ?

Reply:
> [...] we move the pictures to save the inferences but maybe we find another way ?

I dont see if we are checking wether or not the picture has been used in an inference or if its inference has been verified. 

Either way I think it'd be a better practice to move the picture into a dev owned **container** which would contain all the deleted picture with a **verified** inference. 

By moving the deleted picture into a dev owned container this would allow the user to think he has deleted the pictures but we (dev & data analyst) will keep the picture and verified inference for training purposes.

I dont see any other reason to keep a deleted picture otherwise and I dont think a non-verified inference has much value for us to keep. 

WDYT @sylvanie85 @ChromaticPanic

_Note the use case of deleting a folder **with** pictures into it doesn't really fit the average usage of Nachet and the reason why a user would delete a validated seems quite ambiguous_

Reply:
Yes, I completely agree about the dev container.
So the image should be saved if the inference is fully verified, am I right? What if only some boxes are checked? And what about images that come from a batch upload? 

I agree it's an ambiguous use case but as a user can delete a directory using the front end, this could generate errors and it wouldn't make sense if we still had it in the database would it?

And I think it's an interesting use case if we then want to be able to respond to this issue: https://github.com/ai-cfia/nachet-frontend/issues/138

Reply:
Yeah good idea. We should keep the image as long as it has at least 1 verified inference.   
  
Or maybe we should also show further options in the front end when they delete a picture set. "Would you like to remove all verified feedback for this picture set" . Because maybe they uploaded a picture set, they started annotating, but then they realized that the metadata they assigned in the upload form was wrong. ( wrong species, wrong zoom level etc )

Reply:
> Because maybe they uploaded a picture set, they started annotating, but then they realized [...]

If they only started annotating and not every image has been verified, the inference as a whole is not verified. I'd only keep the images with an inference fully verified. Not the partial one.

>Or maybe we should also show further options in the front end when they delete a picture set. "Would you like to remove all verified feedback for this picture set"

I like this idea, however, there are no inferences created from a batch upload. This is another use case of deleting that I think hasn't been covered yet (picture_seed are not deleted). Would it be a hard process to have a back and forth request from the FE-BE to determine if a folder has validated inferences?


_sorry for the poor naming here..._
```mermaid
sequenceDiagram
    participant User
    participant FE
    participant BE
    participant DS

    User->>FE: Delete Folder
    FE->>BE: Request to delete folder
    BE->>DS: Check if there are validated inferences
    DS-->>BE: Validated inference status
    alt Validated inferences exist
        BE->>FE: Request user confirmation
        FE->>User: Ask to keep data for training
        alt No
            User ->>FE: No
            BE->>DS: delete picture_set
        alt YES
            User ->>FE: No
            BE->>DS: save validated inferences in picture_set
            BE->>DS: delete picture_set
    end

```

Reply:
> Would it be a hard process to have a back and forth request from the FE-BE to determine if a folder has validated inferences?

We don't have that yet. but we will need a feature that retrieves "work done" linked to uploaded images, so that previous work can be viewed or continued on the front end. For example classification results, feedback (both complete and incomplete)